lib/forcegraph.js

```diff
@@ -341,21 +345,37 @@ define(["d3"], function (d3) {
       nctx.moveTo(nodeRadius, 0)
       nctx.arc(0, 0, nodeRadius, 0, 2 * Math.PI)
 
-      nctx.strokeStyle = nodeColor
+      nctx.strokeStyle = meshColor
       nctx.lineWidth = nodeRadius
       nctx.stroke()
 
       ctx.save()
       ctx.scale(1 / scale / r, 1 / scale / r)
-      nodes.filter(visibleNodes).forEach(function (d) {
+
+      meshNodes.filter(visibleNodes).forEach(function (d) {
+        ctx.drawImage(node, scale * r * d.x - node.width / 2, scale * r * d.y - node.height / 2)
+      })
+
+      nctx.strokeStyle = uplinkColor
+      nctx.stroke()
+
+      uplinkNodes.filter(visibleNodes).forEach(function (d) {
         ctx.drawImage(node, scale * r * d.x - node.width / 2, scale * r * d.y - node.height / 2)
       })
+
+      nctx.strokeStyle = hotspotColor
+      nctx.stroke()
+
+      hotspotNodes.filter(visibleNodes).forEach(function (d) {
+        ctx.drawImage(node, scale * r * d.x - node.width / 2, scale * r * d.y - node.height / 2)
+      })
+
       ctx.restore()
 
       // -- draw clients --
       ctx.save()
       ctx.beginPath()
-      nodes.filter(visibleNodes).forEach(function (d) {
+      meshNodes.concat(uplinkNodes).concat(hotspotNodes).filter(visibleNodes).forEach(function (d) {
```

Comment:
It may make sense to keep something like `allNodes` around to avoid concatting the arrays again and again.